k256/src/arithmetic.rs

```diff
@@ -124,36 +133,66 @@ impl AffinePoint {
         }
     }
 
+    /// Returns a [`PublicKey`] with the SEC-1 compressed encoding of this point.
+    pub fn to_compressed_pubkey(&self) -> PublicKey {
+        PublicKey::Compressed(self.clone().into())
+    }
+
+    /// Returns a [`PublicKey`] with the SEC-1 uncompressed encoding of this point.
+    pub fn to_uncompressed_pubkey(&self) -> PublicKey {
+        PublicKey::Uncompressed(self.clone().into())
+    }
+}
+
+impl Neg for AffinePoint {
+    type Output = AffinePoint;
+
+    fn neg(self) -> Self::Output {
+        AffinePoint {
+            x: self.x,
+            y: -self.y,
+        }
+    }
+}
+
+impl From<AffinePoint> for CompressedCurvePoint {
     /// Returns the SEC-1 compressed encoding of this point.
-    pub fn to_compressed_pubkey(&self) -> CompressedCurvePoint {
+    fn from(affine_point: AffinePoint) -> CompressedCurvePoint {
         let mut encoded = [0; 33];
-        encoded[0] = if self.y.is_odd().into() { 0x03 } else { 0x02 };
-        encoded[1..33].copy_from_slice(&self.x.to_bytes());
+        encoded[0] = if affine_point.y.is_odd().into() {
+            0x03
+        } else {
+            0x02
+        };
+        encoded[1..33].copy_from_slice(&affine_point.x.to_bytes());
 
         CompressedCurvePoint::from_bytes(GenericArray::clone_from_slice(&encoded[..]))
             .expect("we encoded it correctly")
     }
+}
 
+impl From<AffinePoint> for UncompressedCurvePoint {
     /// Returns the SEC-1 uncompressed encoding of this point.
-    pub fn to_uncompressed_pubkey(&self) -> UncompressedCurvePoint {
+    fn from(affine_point: AffinePoint) -> UncompressedCurvePoint {
         let mut encoded = [0; 65];
         encoded[0] = 0x04;
-        encoded[1..33].copy_from_slice(&self.x.to_bytes());
-        encoded[33..65].copy_from_slice(&self.y.to_bytes());
+        encoded[1..33].copy_from_slice(&affine_point.x.to_bytes());
+        encoded[33..65].copy_from_slice(&affine_point.y.to_bytes());
 
         UncompressedCurvePoint::from_bytes(GenericArray::clone_from_slice(&encoded[..]))
             .expect("we encoded it correctly")
     }
 }
```

Comment:
It seems like if we had an `AffineCoordinates` trait which provided access to the equivalent of `x.to_bytes()` and `y.to_bytes()`, we could have generic `CompressedCurvePoint::from_affine_point` and  `UncompressedCurvePoint::from_affine_point` methods which could DRY out this logic, since it's otherwise identical between the two crates.

If we had that, the bounds used on `FixedBaseScalarMul::Point` could be dramatically simpler since we wouldn't need to bound on the `From` impls.

Reply:
I implemented the above in #50 and it totally works and DRYs out all of this ugly serialization logic into the `elliptic-curve` crate itself so it doesn't have to be reimplemented on a curve-by-curve basis 🎉